apps/src/accounts/AccountInformation/index.tsx

```diff
@@ -0,0 +1,448 @@
+  const [showAccountUpdateSuccess, setShowAccountUpdateSuccess] =
+    useState(false);
+  const [showEmailUpdateSuccess, setShowEmailUpdateSuccess] = useState(false);
+  const [errors, setErrors] = useState<Record<string, string[]>>({});
+
+  const displayNameHelperMessage = useMemo(
+    () =>
+      isStudent ? undefined : i18n.accountInformationDisplayNameHintTeacher(),
+    [isStudent]
+  );
+
+  const emailValue = useMemo(
+    () => (isStudent ? `***${i18n.encrypted()}***` : email ?? ''),
+    [isStudent, email]
+  );
+
+  const usStateOptions = useMemo(
+    () =>
+      [[i18n.accountInformationSelectState(), '']]
+        .concat(usStateDropdownOptions)
+        .map(([text, value]) => ({text, value})),
+    [usStateDropdownOptions]
+  );
+
+  const lockedOutStudentMessage = useMemo(
+    () =>
+      studentInLockoutFlow
+        ? i18n.accountInformationUpdateFieldParentPermissionRequired()
+        : undefined,
+    [studentInLockoutFlow]
+  );
+
+  const handleSubmitAccountSettingsUpdate = async () => {
+    resetMessages();
+    setErrors({});
+    const userUpdates = {
+      name,
+      username,
+      password,
+      password_confirmation: passwordConfirmation,
+      current_password: currentPassword,
+      age: isStudent ? age : '21+',
+      gender_student_input: isStudent && gender ? gender : undefined,
+      us_state: isStudent && isUSA ? usState : undefined,
+      country_code: isStudent ? countryCode : undefined,
+    };
+    const response = await fetch('/users', {
+      method: 'PUT',
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+        'X-CSRF-Token': await getAuthenticityToken(),
+      },
+      body: JSON.stringify({
+        user: userUpdates,
+      }),
+    });
+
+    if (response.ok) {
+      setShowAccountUpdateSuccess(true);
+    } else {
+      const validationErrors = await response.json();
+      setErrors(validationErrors);
+    }
+  };
+
+  const handleSubmitEmailUpdate = async ({
+    newEmail,
+    currentPassword,
+    emailOptIn,
+  }: {
+    newEmail: string;
+    currentPassword: string;
+    emailOptIn: string;
+  }) => {
+    resetMessages();
+    const hashedEmail = hashEmail(newEmail);
+    const response = await fetch('/users/email', {
+      method: 'PATCH',
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+        'X-CSRF-Token': await getAuthenticityToken(),
+      },
+      body: JSON.stringify({
+        user: {
+          email: newEmail,
+          current_password: currentPassword,
+          hashed_email: hashedEmail,
+          email_preference_opt_in: emailOptIn,
+        },
+      }),
+    });
+
+    if (!response.ok) {
+      const validationErrors = await response.json();
+      let error;
+      if (validationErrors) {
+        error = {
+          serverErrors: {
+            newEmail: validationErrors.email && validationErrors.email[0],
```

Comment:
Nit: can this be `validationErrors?.email[0]`?

Reply:
sure! but it will need to be `validationErrors.email?.[0]` I think

Reply:
Oh whoops yes I misread it, whatever syntax does that yes!